From: Joe Perches <joe@perches.com>
Date: Mon, 4 Jun 2012 12:44:18 +0000
Patch-mainline: v3.6-rc1
Subject: drivers: net: Remove casts to same type
Git-commit: c2fd03a0115a244c5f622453b2b1f038ed5700a6

Adding casts of objects to the same type is unnecessary
and confusing for a human reader.

For example, this cast:

        int y;
        int *p = (int *)&y;

I used the coccinelle script below to find and remove these
unnecessary casts.  I manually removed the conversions this
script produces of casts with __force, __iomem and __user.

@@
type T;
T *p;
@@

-       (T *)p
+       p

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: <ohering@suse.de>
---
 drivers/net/appletalk/cops.c      | 2 +-
 drivers/net/can/bfin_can.c        | 2 +-
 drivers/net/can/mcp251x.c         | 3 +--
 drivers/net/fddi/defxx.c          | 4 ++--
 drivers/net/fddi/skfp/pmf.c       | 8 ++++----
 drivers/net/hamradio/mkiss.c      | 8 ++++----
 drivers/net/hyperv/netvsc.c       | 2 +-
 drivers/net/irda/ali-ircc.c       | 6 +++---
 drivers/net/irda/au1k_ir.c        | 2 +-
 drivers/net/slip/slip.c           | 4 ++--
 drivers/net/vmxnet3/vmxnet3_drv.c | 2 +-
 drivers/net/wan/x25_asy.c         | 2 +-
 drivers/net/wimax/i2400m/fw.c     | 2 +-
 13 files changed, 23 insertions(+), 24 deletions(-)

diff --git a/drivers/net/hyperv/netvsc.c b/drivers/net/hyperv/netvsc.c
index 8b91947..4363c76 100644
--- a/drivers/net/hyperv/netvsc.c
+++ b/drivers/net/hyperv/netvsc.c
@@ -614,7 +614,7 @@ retry_send_cmplt:
 static void netvsc_receive_completion(void *context)
 {
 	struct hv_netvsc_packet *packet = context;
-	struct hv_device *device = (struct hv_device *)packet->device;
+	struct hv_device *device = packet->device;
 	struct netvsc_device *net_device;
 	u64 transaction_id = 0;
 	bool fsend_receive_comp = false;
